packages/material-ui/src/TextWidget/TextWidget.tsx

```diff
@@ -43,7 +43,7 @@ const TextWidget = ({
         disabled={disabled || readonly}
         name={name}
         type={schema.type as string}
-        value={value ? value : ''}
+        value={value === undefined || value === null  ? '' : value}
```

Comment:
I believe we would also need to include NaN. Can we do this instead (because I believe the only falsy value we need to include is 0)?

```suggestion
        value={value || (value === 0)  ? '' : value}
```